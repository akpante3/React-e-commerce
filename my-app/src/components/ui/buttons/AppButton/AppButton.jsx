import React from 'react';
import './AppButton.scss';


const AppButton = ({
  title, color, buttonClickHandler, value,
}) => {
  const onClickHandler = () => {
    if (buttonClickHandler) {
      buttonClickHandler(value);
    }
  };

  const data = (
    <button
      type="button"
      className={`app-button app-button-md ${color === 'white' ? 'white-button' : 'pink-button'}`}
      onClick={() => onClickHandler()}
    >
      {title}
    </button>
  );
  return data;
};

export default AppButton;
