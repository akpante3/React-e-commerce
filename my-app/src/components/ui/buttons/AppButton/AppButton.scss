@import '../../../../styles/variables.scss';

.app-button {
	border-radius: 30px;
	font-size: 14px;
	border: none;
	transition: all 0.5s ease 0s;
}
.app-button-md {
	width: 110px;
	padding: 6px 0px;
}
.app-button-lg {
	width: 180px;
	padding: 12px 0px;    
}
.white-button {
	background-color: white ;
	color: #f62e5e;
	border: 1px #f62e5e solid;
}
button:hover {
	background: $dark-pink;
	border: 1px $dark-pink solid;
	color: $white-color;
}
.pink-button {
	background-color: #f62e5e;
	color: white;
	border: 1px #f62e5e solid;
}