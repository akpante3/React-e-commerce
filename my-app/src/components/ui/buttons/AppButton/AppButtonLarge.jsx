import React from 'react';
import './AppButton.scss';


const AppButtonLarge = ({
  title, color, buttonClickHandler, value,
}) => {
  const onClickHandler = () => {
    if (buttonClickHandler) {
      buttonClickHandler(value);
    }
  };

  const data = (
    <button
      type="button"
      className={`app-button app-button-lg ${color === 'white' ? 'white-button' : 'pink-button'}`}
      onClick={() => onClickHandler()}
    >
      {title}

    </button>
  );
  return data;
};

export default AppButtonLarge;
