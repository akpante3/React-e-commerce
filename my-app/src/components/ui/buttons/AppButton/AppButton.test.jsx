import React from 'react';
import { shallow } from 'enzyme';
import AppButton from './AppButton';
import AppLargeButton from './AppButtonLarge';


describe('test for AppButton', () => {
  it('test for appButton', () => {
    const wrapper = shallow(<AppButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it('test for AppLargeButton', () => {
    const wrapper = shallow(<AppLargeButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
