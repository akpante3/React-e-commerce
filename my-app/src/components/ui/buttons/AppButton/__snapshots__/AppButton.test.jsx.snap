// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for AppButton test for AppLargeButton 1`] = `
<button
  className="app-button app-button-lg pink-button"
  onClick={[Function]}
  type="button"
/>
`;

exports[`test for AppButton test for appButton 1`] = `
<button
  className="app-button app-button-md pink-button"
  onClick={[Function]}
  type="button"
/>
`;
